database.go

```diff
@@ -118,22 +121,22 @@ func NewMeasurement(name string) *Measurement {
 	}
 }
 
-// CreateFieldIfNotExists creates a new field with an autoincrementing ID.
+// createFieldIfNotExists creates a new field with an autoincrementing ID.
 // Returns an error if 255 fields have already been created on the measurement.
-func (m *Measurement) createFieldIfNotExists(name string, typ FieldType) (*Field, error) {
+func (m *Measurement) createFieldIfNotExists(name string, typ influxql.DataType) (*Field, error) {
 	// Ignore if the field already exists.
 	if f := m.FieldByName(name); f != nil {
 		return f, nil
 	}
 
 	// Only 255 fields are allowed. If we go over that then return an error.
-	if len(m.Fields) > math.MaxUint8 {
+	if len(m.Fields)+1 > math.MaxUint8 {
 		return nil, ErrFieldOverflow
 	}
 
 	// Create and append a new field.
 	f := &Field{
-		ID:   uint8(len(m.Fields)),
+		ID:   uint8(len(m.Fields) + 1),
```

Comment:
It was a clash with the assumption in `influxql` that a zero field id means that it's missing.